src/basins/edgetracking.jl

```diff
@@ -0,0 +1,194 @@
+Euclidean distance) apart from each other. Next, a `ParallelDynamicalSystem` is initialized
+from these two updated states and integrated forward until the two trajectories diverge
+from each other by more than `eps2` (Euclidean distance). The two final states of the
+parallel integration are then used as new states `u1` and `u2` for a new bisection, and 
+so on, until a stopping criterion is fulfilled.
+
+## Output 
+Output can be controlled via the `output_level` argument.
+* `output_level = 0`: returns only the final state on the edge
+* `output_level = 1`: returns the track along the edge as a vector of states
+* `output_level = 2`: returns `[track, bisect1, bisect2]`, where `track` is the track along  
+  the edge and `bisect1`, `bisect2` are vectors containing the bisection points on either  
+  side of the basin boundary
+
+!!! warning
+    May behave erroneously when the DiffEq solver of `ds` is set to `alg = SimpleATsit5()`,
+    which is the default solver in DynamicalSystems. The recommended solver is `Vern9()`.
+
+## References
+[^1]: [Skufca et al. (2006)](https://journals.aps.org/prl/pdf/10.1103/PhysRevLett.96.174101?casa_token=RUn26KnFdNEAAAAA%3AoXsTlmEWVMkEYbOtR-j2PH2vYOOPOy1a2R_37ncnf4gsiHp6GR66M-IBpzXocLoMQC_oHhk8MIFRa_8)
+[^2]: [Schneider et al. (2008)](https://journals.aps.org/pre/pdf/10.1103/PhysRevE.78.037301?casa_token=mLQmTv_cBGUAAAAA%3AVKnQs290sq1MNm-5k8hW7nJeLtVX54I7l-SEGol_HUPSCwziPi-EGDE8ucrDiVMIXZGUbzzam8benFw)
+[^3]: [Luarini and Bodai (2017)](https://iopscience.iop.org/article/10.1088/1361-6544/aa6b11)
+"""
+function edgetracking(ds::DynamicalSystem, u1, u2, attractors::Dict;
+    eps1=1e-7,
+    eps2=1e-6,
+    maxiter=100,
+    abstol=0.0,
+    tmax=Inf,
+    dt=0.01,
+    ϵ_mapper=0.1,
+    output_level=2,
+    verbose=false,
+    kwargs...)
+    
+    pds = ParallelDynamicalSystem(ds, [u1, u2])
+    mapper = AttractorsViaProximity(ds, attractors, ϵ_mapper; kwargs...)
+    
+    edgetracking(pds, mapper;
+        eps1=eps1, eps2=eps2, maxiter=maxiter, abstol=abstol, dt=dt, tmax=tmax,
+        output_level=output_level, verbose=verbose)
+end;
+
+"""
+    edgetracking(pds::ParallelDynamicalSystem, mapper::AttractorMapper; kwargs...)
+Low-level function for the [`edgetracking`](@ref) algorithm.
+"""
+function edgetracking(pds::ParallelDynamicalSystem, mapper::AttractorMapper;
+    eps1=1e-9,
+    eps2=1e-8,
+    maxiter=100,
+    abstol=0.0,
+    dt=0.01,
+    tmax=Inf,
+    output_level=2,
+    verbose=false)
+    
+    verbose && println("=== Starting edge tracking algorithm ===")
+    
+    u1, u2 = bisect_to_edge(pds, mapper; eps=eps1)
+    edgestate = (u1 + u2)/2
+    
+    if output_level>0
+        bisect_pts1 = [u1]
+        bisect_pts2 = [u2]
+        edge = [edgestate]
+    end
+    verbose && println("... Iteration 1: Edge at $(edgestate)")
+
+    correction = abstol + 1
+    counter = 1
+    # edge track iteration loop
+    while (correction > abstol) & (maxiter > counter)
+        reinit!(pds, [u1,u2])
```

Comment:
Dont do `reinit!(pds, [u1, u2])` becuase this unecessarily does more things than it has to and it allocates a vector when you write `[u1, u2]`. Instead, do
```julia
set_state!(pds, 1, u1)
set_state!(pds, 2, u2)
```